### ISSUE TITLE
Model.create fails with `Cannot read property 'length' of undefined` if there is a second argument with value null or undefined.

### ISSUE THREAD

**Issue by robincsamuel** [Automattic/mongoose#13487]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

7.2.2

### Node.js version

14.20.1

### MongoDB server version

5.0.7

### Description

The function throws the following error when executing `Model.create` with a second argument having value `null` or `undefined`,

```
TypeError: Cannot read property 'length' of undefined
    at Function.create (/Users/robin/Projects/sample/node_modules/mongoose/lib/model.js:2825:14)
```

**It works fine on version `6.x`.**

### Steps to Reproduce

```
const User = model(
  'User',
  new Schema({
    name: String,
  }),
);

const createUser = (payload, options=undefined) => {
   return User.create(payload, options)
}

await createUser({name: "Robin"})

```

### Expected Behavior

The entry should get created. The options parameter should be ignored since it's `undefined`.be ignored

**Comment by vkarpov15** [comment]
We'll fix the unexpected error in #13493, but please be aware that your syntax for `create()` is wrong. Calling `create()` with 2 object arguments creates 2 documents, Mongoose does **not** treat the 2nd arg as `options` **unless** the first arg is an array.

Please use the following instead:

```javascript
const createUser = (payload, options=undefined) => {
   return User.create([payload], options) // <-- array around `payload` to make Mongoose use 2nd arg as options
}
```

**Comment by robincsamuel** [comment]
Great! Ah, thanks for pointing that out. I am actually passing an array but messed up when writing the example.On 09-Jun-2023, at 11:11 PM, Valeri Karpov ***@***.***> wrote:﻿
We'll fix the unexpected error in #13493, but please be aware that your syntax for create() is wrong. Calling create() with 2 object arguments creates 2 documents, Mongoose does not treat the 2nd arg as options unless the first arg is an array.
Please use the following instead:
const createUser = (payload, options=undefined) => {
   return User.create([payload], options) // <-- array around `payload` to make Mongoose use 2nd arg as options
}
